### PR TITLE
Devised

### PR DESCRIPTION
**Pull request by Dmambo**
## FEATURES ADDED

- [x] Removed the  `current_user` method in `ApplicationController`.
- [x] [Install and setup devise](https://github.com/heartcombo/devise#getting-started).

**what was implemented**

1.     Can register a new user.
2.     User logs in with a combination of email and password.
3.     Hashed password should be stored in the database.
4.     Ask for confirmation of email.
5.     Can reset password.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -6,6 +6,10 @@ class User < ApplicationRecord
   validates :name, presence: true, length: { maximum: 50 }
   validates :posts_counter, numericality: { greater_than_or_equal_to: 0, only_integer: true }
 
+  # add devise
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (t-yanick) [inline app/models/user.rb]: - [ ] Great job working on this milestone. For your app to ask confirmation of email, `:confirmable` has to be enabled in the `user` model.
```
	devise :database_authenticatable, :registerable,
				:recoverable, :rememberable, :validatable, :confirmable
```

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi @Dmambo , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Great use of devise for authentication
:heavy_check_mark: Login and registration views same as given wireframe
:heavy_check_mark: Descriptive Readme file and PR

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (Dmambo) [reply app/models/user.rb:11]: @t-yanick 
i did add it to the user using the migrate

email to user

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @Dmambo, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

👍 Great job making the changes requested by the previous reviewer.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**